sdk/synapse/test-resources.json

```diff
@@ -0,0 +1,403 @@
+            "name": "[variables('uniqueDataLakeStorageFilesystemName')]",
+            "properties": {
+                "mode": "Incremental",
+                "template": {
+                    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+                    "contentVersion": "1.0.0.0",
+                    "parameters": {},
+                    "variables": {},
+                    "resources": [
+                        {
+                            "type": "Microsoft.Storage/storageAccounts/blobServices/containers",
+                            "name": "[concat(variables('uniqueDataLakeStorageAccountName'), '/default/', variables('uniqueDataLakeStorageFilesystemName'))]",
+                            "apiVersion": "2018-02-01",
+                            "properties": {
+                                "publicAccess": "None"
+                            }
+                        }
+                    ]
+                }
+            },
+            "subscriptionId": "[parameters('storageSubscriptionID')]",
+            "resourceGroup": "[parameters('storageResourceGroupName')]",
+            "condition": "[parameters('isNewFileSystemOnly')]"
+        },
+        {
+            "type": "Microsoft.Synapse/workspaces/bigDataPools",
+            "apiVersion": "2019-06-01-preview",
+            "name": "[concat(variables('uniqueWorkspaceName'), '/', variables('uniqueSparkPoolName'))]",
+            "location": "[parameters('location')]",
+            "tags": "[parameters('tags')]",
+            "properties": {
+                "nodeCount": "[parameters('nodeCount')]",
+                "nodeSizeFamily": "[parameters('nodeSizeFamily')]",
+                "nodeSize": "[parameters('nodeSize')]",
+                "autoScale": {
+                    "enabled": "[parameters('autoScaleEnabled')]",
+                    "minNodeCount": "[parameters('autoScaleMinNodeCount')]",
+                    "maxNodeCount": "[parameters('autoScaleMaxNodeCount')]"
+                },
+                "autoPause": {
+                    "enabled": "[parameters('autoPauseEnabled')]",
+                    "delayInMinutes": "[parameters('autoPauseDelayInMinutes')]"
+                },
+                "sparkVersion": "[parameters('sparkVersion')]",
+                "libraryRequirements": {
+                    "filename": "[parameters('libraryRequirementsFilename')]",
+                    "content": "[parameters('libraryRequirementsContent')]"
+                }
+            }
+        }
+    ],
+    "outputs": {
+        "TENANT_ID": {
+            "type": "string",
+            "value": "[parameters('tenantId')]"
+        },
+        "CLIENT_ID": {
+            "type": "string",
+            "value": "[parameters('testApplicationId')]"
+        },
+        "CLIENT_SECRET": {
+            "type": "string",
+            "value": "[parameters('testApplicationSecret')]"
+        },
```

Comment:
You shouldn't need these as the New-TestResources sets them automatically.

Reply:
And once you remove those, see if you can remove some of those parameters mentioned above as well. Should make it easier to maintain.